src/interner.rs

```diff
@@ -26,24 +27,30 @@ use {crate::parking_lot_shim::*, parking_lot::RwLock};
 /// uses raw-pointer borrowing rules to avoid invalidating extant references.
 ///
 /// The resolved reference is guaranteed valid until the PinBox is dropped.
-struct PinBox<T: ?Sized>(NonNull<T>);
+struct PinBox<T: ?Sized> {
+    ptr: NonNull<T>,
+    _marker: PhantomData<T>,
```

Comment:
```suggestion
    _marker: PhantomData<Box<T>>,
```

Is there a specific goal of adding the `PhantomData` here, though? I agree it's better because it's "by example"ing the semantics of the pointer, but as I recall there's no actual impact to holding the phantom here, since `NonNull<T>` is already covariant over `T`.

Reply:
I've switched the `PhantomData<T>` to `PhantomData<Box<T>>` in f516a683df84ddd50369015dd9b933ccda8c50e1 as you suggested.

I was going by https://github.com/rust-lang/rfcs/blob/master/text/0769-sound-generic-drop.md#phantom-data when including the `PhantomData` to ensure that `PinBox<T>` owns a `T` with regards to dropck.

Reply:
That RFC was superceded by the `#[may_dangle]` attribute rather than inferring, IIUC, and the only thing `PhantomData` impacts is variance and autotraits.

That said, though, the "by example" PhantomData is not a bad thing.

Reply:
Thanks for explaining that!